examples/java-springboot/src/main/java/com/spruceid/didkitexample/util/VerifiablePresentation.java

```diff
@@ -3,106 +3,209 @@
             final Map<String, Object> resultMap = mapper.readValue(result, new TypeReference<>() {
             });
 
             if (((List<String>) resultMap.get("errors")).size() > 0) {
-                System.out.println("[ERROR] VC: " + resultMap.get("errors"));
-                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid credential");
+                logger.error("VC: " + resultMap.get("errors"));
+                throw new ResponseStatusException(
+                              HttpStatus.BAD_REQUEST,
+                              "Invalid credential"
+                          );
             }
         } catch (Exception e) {
-            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Failed to verify credential");
+            logger.error("Exception validating credential: " + e);
+            throw new ResponseStatusException(
+                          HttpStatus.INTERNAL_SERVER_ERROR,
+                          "Failed to verify credential"
+                      );
         }
 
         return vc;
     }
 
-    private static Map<String, Object> getFirstVc(Object vcs) {
 
+    public static void validateCreatedTimes(
+        @NonNull final Instant now,
+        @NonNull final Duration maxClockSkew,
+        @NonNull final Map<String, Object> presentation
+    ) throws Exception {
+        final Instant credentialCreated = getCredentialCreated(presentation);
+        final Instant presentationCreated = getPresentationCreated(presentation);
+        final Instant nowWithSkew = now.plus(maxClockSkew);
+
+        if(credentialCreated.compareTo(nowWithSkew) > 0) {
+            logger.error("The Credential in the presentation is not yet valid");
+            logger.error("credentialCreated: " + credentialCreated);
+            logger.error("processedAt:       " + nowWithSkew);
+            throw new ResponseStatusException(
+                          HttpStatus.BAD_REQUEST,
+                          "Credential in presentation is not yet valid"
+                      );
+        }
+
+        if(presentationCreated.compareTo(nowWithSkew) > 0) {
+            logger.error("The presentation is not yet valid");
+            logger.error("presentationCreated: " + presentationCreated);
+            logger.error("processedAt:         " + nowWithSkew);
+            throw new ResponseStatusException(
+                          HttpStatus.BAD_REQUEST,
+                          "Presentation is not yet valid"
+                      );
+        }
+    }
+
+    public static Instant getCredentialCreated(
+        @NonNull final Map<String, Object> presentation
+    ) {
+        final Object vcs = presentation.get("verifiableCredential");
+        final Map<String, Object> vc = getFirstVc(vcs);
+        final Map<String, Object> proof = (Map<String, Object>)vc.get("proof");
+        final String createdStr = (String)proof.get("created");
```

Comment:
You don't want to create const Strings for these constants like "created"?

Sometimes people use an enum type for a set of related String constants. See https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html